src/rounded_rect.rs

```diff
@@ -0,0 +1,305 @@
+//! A rectangle with rounded corners.
+
+use crate::{arc::ArcAppendIter, Arc, PathEl, Point, Rect, Shape, Vec2};
+use std::f64::consts::{FRAC_PI_2, PI};
+
+/// A rectangle with equally rounded corners.
+///
+/// By construction the rounded rectangle will have
+/// non-negative dimensions and radius clamped to half size of the rect.
+#[derive(Clone, Copy, Default, Debug)]
+pub struct RoundedRect {
+    /// Coordinates of the rectangle.
+    rect: Rect,
+    /// Radius of all four corners.
+    radius: f64,
+}
+
+impl RoundedRect {
+    /// A new rectangle from minimum and maximum coordinates.
+    #[inline]
+    pub fn new(x0: f64, y0: f64, x1: f64, y1: f64, radius: f64) -> RoundedRect {
```

Comment:
Two thoughts:

- I would add a new constructor that takes a `Rect` and a `radius`.
- In `RoundedRect::new`, I would just call `Rect::new`, and then I would call the new constructor mentioned above.

Reply:
Thanks, done!